x/concentrated-liquidity/math/math.go

```diff
@@ -26,6 +27,7 @@ func Liquidity0(amount osmomath.Int, sqrtPriceA, sqrtPriceB osmomath.BigDec) osm
 		panic(fmt.Sprintf("liquidity0 diff is zero: sqrtPriceA %s sqrtPriceB %s", sqrtPriceA, sqrtPriceB))
 	}
 
+	// TODO (perf): consider Dec() function that does not reallocate
 	return amountBigDec.MulMut(product).QuoMut(diff).Dec()
```

Comment:
Note: this should be addressed by https://github.com/osmosis-labs/osmosis/pull/6261, CC: @pysel